scripts/prepare_disk.sh

```diff
@@ -128,6 +176,16 @@ if [ ! -r "${PARTITION_TABLE_FILE}" ]; then
     exit 1
 fi
 
+if [ ! -b "${TARGET_DISK}" ]; then
+    echo "Error, block device '${TARGET_DISK}' does not exist."
+    exit 1
+fi
+
+if ! is_resize_needed; then
```

Comment:
be very wary that this will 'break' set -e

Reply:
So we have to be very sparing in using this construct. (I'm not sure what happens if you call set -e in the function itself again, but that's more of a patchwork).

In this function, I think we are good as we pretty much check each and every call, except the cleanup unlink.

The only solution I can come up with however, is:
```sh
resize_needed="$(is_resize_needed; echo "${?}")"
if [ "${resize_needed}" -eq 1 ]; then
    echo "Partition resize not required."
    exit 0
fi
```
Which also looks a little quirky. But I suppose we cannot get around quirks with the tools we have right now.